### PR TITLE
WES analysis support 

### PR DESCRIPTION
**Pull request by roshni-b**
* Support for the following WES Analysis Files (see [wes_analysis_template.json](https://github.com/CIMAC-CIDC/cidc-schemas/blob/wes_analysis_separate_from_assay/cidc_schemas/schemas/templates/analyses/wes_analysis_template.json))

├ analysis

├── align

│   ├── CA209009-Normal
│   │   ├── CA209009-Normal.sorted.bam
│   │   ├── CA209009-Normal.sorted.bam.bai
│   │   ├── CA209009-Normal.sorted.dedup.bam
│   │   ├── CA209009-Normal.sorted.dedup.bam.bai
│   │   ├── CA209009-Normal_recalibrated.bam
│   │   └── CA209009-Normal_recalibrated.bam.bai

│   ├── CA209009-Tumor
│   │   ├── CA209009-Tumor.sorted.bam
│   │   ├── CA209009-Tumor.sorted.bam.bai
│   │   ├── CA209009-Tumor.sorted.dedup.bam
│   │   ├── CA209009-Tumor.sorted.dedup.bam.bai
│   │   ├── CA209009-Tumor_recalibrated.bam
│   │   └── CA209009-Tumor_recalibrated.bam.bai

├── copynumber
│   └── CA209009_Run
│       ├── CA209009_Run_cnvcalls.txt
│       └── CA209009_Run_cnvcalls.txt.tn.tsv

├── corealignments
│   └── CA209009_Run
│       ├── CA209009_Run_tn_corealigned.bam
│       └── CA209009_Run_tn_corealigned.bam.bai

├── germline
│   └── CA209009_Run
│       └── CA209009_Run_vcfcompare.txt

├── metrics

│   ├── CA209009-Normal
│   │   ├── CA209009-Normal.broad.mosdepth.region.dist.txt
│   │   ├── CA209009-Normal.mda.mosdepth.region.dist.txt
│   │   ├── CA209009-Normal.mocha.mosdepth.region.dist.txt
│   │   ├── CA209009-Normal_coverage_metrics.txt
│   │   ├── CA209009-Normal_coverage_metrics.txt.sample_summary
│   │   ├── CA209009-Normal_target_metrics.txt
│   │   └── CA209009-Normal_target_metrics.txt.sample_summary

│   ├── CA209009-Tumor
│   │   ├── CA209009-Tumor.broad.mosdepth.region.dist.txt
│   │   ├── CA209009-Tumor.mda.mosdepth.region.dist.txt
│   │   ├── CA209009-Tumor.mocha.mosdepth.region.dist.txt
│   │   ├── CA209009-Tumor_coverage_metrics.txt
│   │   ├── CA209009-Tumor_coverage_metrics.txt.sample_summary
│   │   ├── CA209009-Tumor_target_metrics.txt
│   │   └── CA209009-Tumor_target_metrics.txt.sample_summary
│   └── all_sample_summaries.txt

├── neoantigen
│   └── CA209009_Run

│       └── MHC_Class_I
│           ├── CA209009-Tumor.all_epitopes.tsv
│           └── CA209009-Tumor.filtered.condensed.ranked.tsv

│    └── MHC_Class_2
│           ├── CA209009-Tumor.all_epitopes.tsv
│           └── CA209009-Tumor.filtered.condensed.ranked.tsv

├── optitype
│   ├── CA209009-Normal
│   │   ├── CA209009-Normal_result.tsv
│   │   └── CA209009_report-{CA209009}-hla.json
│   └── CA209009-Tumor
│   │   ├── CA209009-Tumor_result.tsv
│   │   └── CA209009_report-{CA209009}-hla.json

├── purity
│   └── CA209009_Run
│       └── CA209009_Run_purity_results.txt

└── somatic
    ├── CA209009_Run
    │   ├── CA209009_Run_tnscope.filter.exons.broad.vcf.gz
    │   ├── CA209009_Run_tnscope.filter.exons.mda.vcf.gz
    │   ├── CA209009_Run_tnscope.filter.exons.mocha.vcf.gz
    │   ├── CA209009_Run_tnsnv.filter.exons.broad.vcf.gz
    │   ├── CA209009_Run_tnsnv.filter.exons.mda.vcf.gz
    │   ├── CA209009_Run_tnsnv.filter.exons.mocha.vcf.gz
    │   ├── CA209009_Run_tnhaploytper2.filter.exons.broad.vcf.gz
    │   ├── CA209009_Run_tnhaploytper2.filter.exons.mda.vcf.gz
    │   ├── CA209009_Run_tnhaploytper2.filter.exons.mocha.vcf.gz
    │   ├── CA209009_Run_tnscope.filter.maf
    │   ├── CA209009_Run_tnscope.filter.vcf
    │   ├── CA209009_Run_tnscope.output.maf
    │   ├── CA209009_Run_tnscope.output.vcf
    │   ├── CA209009_Run_tnsnv.filter.maf
    │   ├── CA209009_Run_tnsnv.filter.vcf
    │   ├── CA209009_Run_tnsnv.output.maf
    │   ├── CA209009_Run_tnsnv.output.vcf
    │   ├── CA209009_Run_tnhaploytper2.filter.maf
    │   ├── CA209009_Run_tnhaploytper2.filter.vcf
    │   ├── CA209009_Run_tnhaploytper2.output.maf
   └  └── CA209009_Run_tnhaploytper2.output.vcf

* Updated docs
* Corresponding artifacts updates
* Corresponding prism tests ([test_prism.py](https://github.com/CIMAC-CIDC/cidc-schemas/blob/wes_analysis_separate_from_assay/tests/test_prism.py))

### CODE REVIEW

File: tests/test_prism.py [hunk 1]
```diff
@@ -1308,16 +1308,226 @@ def test_prism_many_artifacts_from_process_as_on_one_record(monkeypatch):
                                     "type": "string",
                                     "process_as": [
                                         {
-                                            "parse_through": "lambda x: f'analysis/sorted/{x}/{x}.output.vcf'",
-                                            "merge_pointer": "/somatic_vcf",
-                                            "gcs_uri_format": "{run_id}/somatic.vcf",
+                                            "parse_through": "lambda x: f'analysis/germline/{x}/{x}_vcfcompare.txt'",
```
Comment (curlup) [inline tests/test_prism.py]: wait. Why do we need all that? 
It's a test that mocks *a part of* wes analysis structure to test the ability of the prism to parse templates formatted in that way. *It is not* the actual wes_analysis template test - so no need here for all 100+ entries etc: it makes test sorta unreadable, don't you think?

File: cidc_schemas/prism.py [hunk 2]
```diff
@@ -35,33 +35,29 @@ def _set_val(
     This function given a *pointer* (jsonpointer RFC 6901 or relative json pointer)
```
Comment (curlup) [inline cidc_schemas/prism.py]: Is all that reformatting somehow related to wes analysis?

Comment (roshni-b) [reply cidc_schemas/prism.py]: Not related to WES at all. Happened automatical with black 🤷

File: cidc_schemas/schemas/assays/components/somatic.json [hunk 3]
```diff
@@ -43,10 +40,57 @@
...
+      "$ref": "artifacts/artifact_gz.json"
+    },
+    "gz_tnscope_exons_mocha": {
+      "$ref": "artifacts/artifact_gz.json"
+    },
+    "gz_tnhaploytper2_exons_mocha": {
+      "$ref": "artifacts/artifact_gz.json"
+    },
+    "gz_tnsnv_exons_mocha": {
+      "$ref": "artifacts/artifact_gz.json"
```
Comment (curlup) [inline cidc_schemas/schemas/assays/components/somatic.json]: I suggest we use something like `artifact_vcf_gz`.

Maybe drop the `gz_` prefix?

File: cidc_schemas/schemas/templates/analyses/wes_analysis_template.json [hunk 4]
```diff
@@ -23,19 +23,229 @@
...
+                                {
+                                    "parse_through": "lambda run: f'analysis/somatic/{run}/{run}_tnsnv.output.maf'",
+                                    "merge_pointer": "/somatic/maf_tnsnv_output",
+                                    "gcs_uri_format": "{run id}/wes_analysis/maf_tnsnv_output.maf",
+                                    "type_ref": "assays/components/local_file.json#properties/file_path",
+                                    "is_artifact": 1
+                                },
+                                {
+                                    "parse_through": "lambda run: f'analysis/somatic/{run}/{run}_tnscope.filter.vcf'",
+                                    "merge_pointer": "/somatic/vcf_tnscope_filtered",
```
Comment (curlup) [inline cidc_schemas/schemas/templates/analyses/wes_analysis_template.json]: maybe rename `filtered` to `filter` as in the output and in the gcs

File: cidc_schemas/schemas/assays/components/ngs/wes_pair_analysis.json [hunk 5]
```diff
@@ -33,6 +33,9 @@
     },
     "clonality": {
       "$ref": "assays/components/clonality.json"
+    },
+    "corealignments": {
+      "$ref": "assays/components/corealignments.json"
     }
   },
   "required": [
```
Comment (curlup) [inline cidc_schemas/schemas/assays/components/ngs/wes_pair_analysis.json]: add additionalProperties: false?

File: cidc_schemas/schemas/assays/components/somatic.json [hunk 6]
```diff
@@ -5,9 +5,6 @@
   "type": "object",
   "description": "Somatic variant calling output sample files.",
   "properties": {
-    "vcf_filtered_compressed": {
-      "$ref": "artifacts/artifact_gz.json"
-    },
```
Comment (curlup) [inline cidc_schemas/schemas/assays/components/somatic.json]: was that the only one not in the `required` ?

**Pull request merged**